nexus/src/app/background/sync_switch_configuration.rs

```diff
@@ -817,6 +999,75 @@ impl BackgroundTask for SwitchPortSettingsManager {
                             .unwrap_or(SwitchLinkSpeed::Speed100G)
                             .into(),
                     };
+
+                    for peer in port_config.bgp_peers.iter_mut() {
+                        peer.communities = match self
+                            .datastore
+                            .communities_for_peer(
+                                opctx,
+                                port.port_settings_id.unwrap(),
+                                &peer.port,
+                                IpNetwork::from(IpAddr::from(peer.addr))
+                            ).await {
```

Comment:
same comments here, should await in parallel most likely.

Reply:
Added TODOs in the code and would like to leave this for a follow on,